lib/Adapter/TolerantParser/Fixer/IndentationFixer.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Phpactor\CodeBuilder\Adapter\TolerantParser\Fixer;
+
+use Microsoft\PhpParser\Node;
+use Microsoft\PhpParser\Node\ClassMembersNode;
+use Microsoft\PhpParser\Node\DelimitedList;
+use Microsoft\PhpParser\Node\Statement\CompoundStatementNode;
+use Microsoft\PhpParser\Parser;
+use Phpactor\CodeBuilder\Domain\TextEdit;
+use Phpactor\CodeBuilder\Domain\StyleProposer;
+use Phpactor\CodeBuilder\Domain\TextEdits;
+use Phpactor\CodeBuilder\Util\TextFormat;
+use Phpactor\CodeBuilder\Util\TextUtil;
+
+class IndentationFixer implements StyleProposer
+{
+    /**
+     * @var Parser
+     */
+    private $parser;
+
+    /**
+     * @var string
+     */
+    private $indent;
+
+    public function __construct(Parser $parser, string $indent = '    ')
+    {
+        $this->parser = $parser;
+        $this->indent = $indent;
+    }
+
+    public function propose(string $text): TextEdits
+    {
+        $node = $this->parser->parseSourceFile($text);
+        $edits = $this->indentations($node, 0);
+
+        return TextEdits::fromTextEdits($edits);
+    }
+
+    private function indentations(Node $node, int $level): array
+    {
+        $edits = [];
+
+        $edits = $this->indent(
+            $edits,
+            $node,
+            $level
+        );
+
+        if (
+            $node instanceof ClassMembersNode ||
+            $node instanceof CompoundStatementNode ||
+            $node instanceof DelimitedList
+        ) {
+            $level++;
+        }
+
+        foreach ($node->getChildNodes() as $childNode) {
+            $edits = array_merge($edits, $this->indentations($childNode, $level));
+        }
+
+        return $edits;
+    }
+
+    private function indent(array $edits, Node $node, int $level): array
+    {
+        $text = $node->getFileContents();
+        $start = $node->getFullStart();
+        $length = $this->getOffsetUntilFirstChild($node) - $start;
+
+        if ($length === 0) {
+            return $edits;
+        }
+
+        $text = substr($text, $start, $length);
+
+        // if there are no new lines in the selection, return
+        if (count(TextUtil::lines($text)) === 1) {
+            return $edits;
+        }
+
+        $text = TextFormat::indentationRemove($text);
+        $text = TextFormat::indentApply($text, $this->indent, $level);
```

Comment:
I see that for the second time, it might be interesting to add something like `TextFormat::indentReplace`

Reply:
Done